packages/astro/src/vite-plugin-astro-server/base.ts

```diff
@@ -23,8 +24,8 @@ export function baseMiddleware(
 
 	return function devBaseMiddleware(req, res, next) {
 		const url = req.url!;
-		if (manySlashes.test(url)) {
-			const destination = url.replace(manySlashes, '/');
+		const destination = collapseDuplicateTrailingSlashes(url, true);
+		if (url && destination !== url) {
 			return writeRedirectResponse(res, 301, destination);
 		}
```

Comment:
I think I'd move this to a its own middleware since it's no longer about the dev base handling, but not a big blocker for the PR.

Reply:
I didn't even notice that that was why it was called base middleware!